substrate/bin/node/runtime/src/lib.rs

```diff
@@ -2009,84 +2008,169 @@ impl pallet_broker::Config for Runtime {
 	type PriceAdapter = pallet_broker::Linear;
 }
 
-construct_runtime!(
-	pub struct Runtime
-	{
+#[frame_support::construct_runtime_v2]
+mod runtime {
+	#[frame::runtime]
+	pub struct Runtime;
+
+	#[frame::pallets]
+	#[frame::derive(
+		RuntimeCall,
+		RuntimeEvent,
+		RuntimeError,
+		RuntimeOrigin,
+		RuntimeFreezeReason,
+		RuntimeHoldReason,
+		RuntimeSlashReason,
+		RuntimeLockId
+	)]
+	pub struct Pallets {
+		#[frame::pallet_index(0)]
```

Comment:
this gets pretty verbose imo... maybe we just bake the index into the RHS?

```rust
pub struct Pallets {
    System: (frame_system, 0),
    Utility: (pallet_utility, 1),
    Timestamp: (pallet_timestamp, 3),
    Babe: (pallet_babe, 2),
    ...
}
```

Reply:
thoughts @kianenigma?

Reply:
The major reason to do this is to maintain a valid Rust syntax while supporting the intermediate expansion arising from `tt_default_parts`.

Essentially, the syntax you suggested is not a valid Rust syntax. Even if we use an enum or a way to make this one work, during the expansion of this macro, `System: frame_system` first becomes `System: frame_system + Pallet + Call ..` where the added tokens are coming from the default parts defined by each pallet. This intermediate expansion should still stay a valid rust syntax for the macro to work.

Another reason in my mind was to make things explicit, similar to what we do for `call_index` instead of trying to build up a strange pattern. Happy to discuss this further, though.

Reply:
Also find it verbose, but at the same time it is a common rust patter to annotate struct fields like this. 

Does the intermediate code HAVE TO be valid rust code, is this a self-inflicted restricton?

Reply:
It actually comes from the restriction in procedural macros. The intermediate code adds all pallet parts, which then again needs to be parsed by the macro.

Reply:
I think maybe the real issue here is the attributes are just so long with the `pallet::` prefix... if it was more like derive macro helpers and something short like `#[index(3)]` maybe it would look less daunting

Reply:
gav's suggestion is better though 👍🏻